src/dags/common/dataschema.py

```diff
@@ -0,0 +1,50 @@
+import json, jsonschema
+from .utils import schema_fetch_url_file
+from schematools.provenance.create import ProvenaceIteration
+
+
+def get_column_names_for_ogr2ogr_sql_select(
+    dataschema, file_to_save, source_filename, source_missing_cols=(), skip_cols=()
+):
+    """Creates a string to use in ogr2ogr bash cmd within the SQL select option (-sql) \
+        It returns the renamed source columns as alias for the orginal (which is stated in the provenance element in the datasetschema) 
+        I.e. orginal_column_name AS renamed_column_name, [orginal_column_name AS renamed_column_name]
+        """
+    # get the data as URL or File
+    data = schema_fetch_url_file(dataschema)
+
+    # get the table columns with provenance defined in schema
+    try:
+        instance = ProvenaceIteration(data)
+        data = instance.final_dic
+
+    except (jsonschema.ValidationError, jsonschema.SchemaError, KeyError) as e:
+        print(str(e), err=True)
+        exit(1)
+
+    # process data to get the column namens and its alias
+    try:
+        output = "select "
```

Comment:
In output wordt de definitieve string "verzameld" door er telkens stukjes aan te concateneren. Dat is op zich prima. Maar, zeker in python, minder efficient. Nu zal dat hier niet zoveel uitmaken. De meer 'pythonic' manier om dit te doen is een list maken waarin de stukjes string worden verzameld. Op het laatste kun je die dan in een keer tot een string omvormen, als volgt:

    output = ['aa', 'bb', 'cc']
    " ".join(output)   # geeft "aa bb cc"

Reply:
Snap wat je bedoeld. Ik heb aangepast. De code is nu iets 'leaner'. Ik had wel twee .joins nodig. 1 voor comma gescheiden colommen en ik een ander nog gebruikt voor spatie gescheiden join van het keyword SELECT en FROM. Deze moesten natuurlijk niet door een comma gescheiden worden gejoind.